### PR TITLE
[GPU] Set affinity TBB using CPUStreamsExecutor

### PR DESCRIPTION
**Pull request by ahnyoung-paul**
### Details:
 - PLEASE DO NOT REVIEW YET.
 - *Set TBB preferred core type by IE config*
   - *Need to rebase code after #7688 will be merged*
   - *Need to make decision for GPU_PLUGIN_PRIORITY usage*
   - *Need to update documentation*

### Tickets:
 - *63585*

### CODE REVIEW

File: docs/IE_DG/supported_plugins/GPU.md [hunk 1]
```diff
@@ -105,15 +105,15 @@ When specifying key values as raw strings (that is, when using Python API), omit
 | `KEY_CACHE_DIR`      | `"<cache_dir>"`                    | `""`              | Specifies a directory where compiled OCL binaries can be cached. First model loading generates the cache, and all subsequent LoadNetwork calls use precompiled kernels which significantly improves load time. If empty - caching is disabled             |
 | `KEY_PERF_COUNT`      | `YES` / `NO`                    | `NO`              | Collect performance counters during inference             |
 | `KEY_CONFIG_FILE`     | `"<file1> [<file2> ...]"`         | `""`              | Load custom layer configuration files                     |
-| `KEY_GPU_PLUGIN_`<br>`PRIORITY` | `<0-3>`                       | `0`               | OpenCL queue priority (before usage, make sure your OpenCL driver supports appropriate extension)<br> Higher value means higher priority for OpenCL queue. 0 disables the setting. |
```
[the remark this answers is not in the document]
Comment (ahnyoung-paul) [reply docs/IE_DG/supported_plugins/GPU.md:108]: Fixed to MODEL_PRIORITY and deprecated PLUGIN_PRIORITY. thanks

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp [hunk 2]
```diff
@@ -32,22 +34,32 @@ void compile_graph::run(program& p) {
...
+    compile_graph_confilg._name = "CLDNNPlugin executor for compile graph on load network";
+    compile_graph_confilg._threadsPerStream = p.get_engine().get_device_info().supports_immad ? 1: compile_graph_confilg._threadsPerStream;
+
+    auto task_executor = std::unique_ptr<CPUStreamsExecutor>(new CPUStreamsExecutor(compile_graph_confilg));
+
+    std::exception_ptr exception;
+    auto compile_graph_func = [&] {
+        try {
+            auto& proc_order = p.get_processing_order();
+            tbb::parallel_for(tbb::blocked_range<size_t>(0, proc_order.size()), [&proc_order, &p](const tbb::blocked_range<size_t>& r) {
```
[the remark this answers is not in the document]
Comment (ahnyoung-paul) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp]: @myshevts I fixed codes to sue only task executor without TBB.
Could you please review it again? thanks.

File: docs/IE_DG/supported_plugins/GPU.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahnyoung-paul) [reply docs/IE_DG/supported_plugins/GPU.md:108]: @vladimir-paramuzov Could you please review it again? thanks.

**Comment by myshevts** [comment]
having no direct TBB and associated dependecies is also great

**Review by myshevts — APPROVED** [review]

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp [hunk 3]
```diff
@@ -14,12 +14,10 @@
...
 #include <iomanip>
 
-#if (CLDNN_THREADING == CLDNN_THREADING_TBB)
-#include <tbb/parallel_for.h>
-#include <tbb/blocked_range.h>
-#endif
+#include <threading/ie_cpu_streams_executor.hpp>
 
 using namespace cldnn;
+using namespace InferenceEngine;
```
Comment (ahnyoung-paul) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp:20]: remove namespace

File: inference-engine/src/cldnn_engine/cldnn_remote_context.cpp [hunk 4]
```diff
@@ -251,16 +251,19 @@ CLDNNExecutionContextImpl::CLDNNExecutionContextImpl(const std::shared_ptr<IInfe
             queue_type = cldnn::queue_types::out_of_order;
         }
 
+
+        IStreamsExecutor::Ptr task_executor = std::make_shared<CPUStreamsExecutor>(m_config.task_exec_config);
```
Comment (ahnyoung-paul) [inline inference-engine/src/cldnn_engine/cldnn_remote_context.cpp:255]: replace to ITaskExecutor

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp (hunk 3 above)
Comment (ahnyoung-paul) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/compile_graph.cpp:20]: Fixed it.

File: inference-engine/src/cldnn_engine/cldnn_remote_context.cpp (hunk 4 above)
Comment (ahnyoung-paul) [reply inference-engine/src/cldnn_engine/cldnn_remote_context.cpp]: Fixed it.

File: docs/IE_DG/supported_plugins/GPU.md [hunk 5]
```diff
@@ -105,15 +105,16 @@ When specifying key values as raw strings (that is, when using Python API), omit
...
.... For hybrid processors, if the task priority type is set as `HIGH` the task will have higher priority for core type selection, and vice versa. If the host processor is not hybrid core or the multi threading is not using tbb, it is set as `ANY`, which is the default type.<br/><br/>OpenCL queue priority is specified by `GPU_QUEUE_PRIORITY_[level]` and there are four types of levels: `HIGH`, `MED`, `LOW`, and `DEFAULT`, where the default value is `DEFAULT`. Before usage, make sure your OpenCL driver supports appropriate extension.<br/><br/>Basically `GPU_MODEL_PRIORITY` can be set as combination of the two priority types, such as<br/>-`GPU_QUEUE_PRIORITY_HIGH\|GPU_HOST_TASK_PRIORITY_HIGH` or<br/>-`GPU_QUEUE_PRIORITY_LOW\|GPU_HOST_TASK_PRIORITY_HIGH`.<br/><br/>Also it can be set as a more abstract level of priority PLUGIN_PRIORIY_[level], which represents combination of the two priorities as follows:<br/>-`GPU_MODEL_PRIORITY_HIGH` : `GPU_QUEUE_PRIORITY_HIGH\|GPU_HOST_TASK_PRIORITY_HIGH`<br/>-`GPU_MODEL_PRIORITY_LOW` : `GPU_QUEUE_PRIORITY_LOW\|GPU_HOST_TASK_PRIORITY_LOW`<br/><br/>The default of `KEY_GPU_MODEL_PRIORITY` is `GPU_QUEUE_PRIORITY_DEFAULT\|GPU_HOST_TASK_PRIORITY_ANY`.<br>  |
```
Comment (vladimir-paramuzov) [inline docs/IE_DG/supported_plugins/GPU.md:108]: `GPU_QUEUE_PRIORITY_<LOW\|HIGH\|MED\|DISABLED>` -> `GPU_QUEUE_PRIORITY_<LOW\|HIGH\|MED\|DEFAULT>`

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM. Please fix minor issue in docs (can be done separately)

File: docs/IE_DG/supported_plugins/GPU.md (hunk 5 above)
Comment (ahnyoung-paul) [reply docs/IE_DG/supported_plugins/GPU.md:108]: thank you for your comments, I forget to fix it. I will update it next PR.

Comment (ahnyoung-paul) [reply docs/IE_DG/supported_plugins/GPU.md:108]: @vladimir-paramuzov Could you please merge this PR? thanks.

**Pull request merged**